discord/installer/install_linux.sh

```diff
@@ -1,60 +1,67 @@
 #!/bin/bash
-DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
-INSTALL_OPTIONS=("Download correct node and install" "Install using local version" "Exit")
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )";
+INSTALL_OPTIONS=("Download correct node and install" "Install using local version" "Exit");
+DISCORD_TEST="resources/app.asar";
 
 runInstaller() {
     echo "Please enter the install directory";
+    
+    #Get and verify install dir
     read INSTALL_DIR;
-    echo "Installing to $INSTALL_DIR";
-
-    $DIR/node $DIR/index.js $DIR -p $INSTALL_DIR;
+    if [ -e "${INSTALL_DIR}/${DISCORD_TEST}"  ]; then
+        echo "Installing to $INSTALL_DIR";
+        $1 $DIR/index.js $DIR -p $INSTALL_DIR;
+    else
+        echo "Couldn't find ${INSTALL_DIR}/${DISCORD_TEST}, please enter a different directory.";
+    fi
 }
 
-#TODO:  Confirm this is cool with the user
 downloadNodeAndInstall() {
-    echo "Downloading standalone tarball..." 
-    if [ `getconf LONG_BIT` = "64" ]; then
-        curl -o $DIR/node.tar.gz https://nodejs.org/dist/v4.2.4/node-v4.2.4-linux-x64.tar.gz
-        echo "Unpacking node..."
-        tar --strip-components=2 -xvf $DIR/node.tar.gz -C $DIR node-v4.2.4-linux-x64/bin/node
+    echo "Downloading standalone tarball...";
+    if [ $(getconf LONG_BIT) == "64" ]; then
+        #Download 64-bit node
+        curl -o $DIR/node.tar.gz https://nodejs.org/dist/v4.2.4/node-v4.2.4-linux-x64.tar.gz;
+        echo "Unpacking node...";
+        tar --strip-components=2 -xvf $DIR/node.tar.gz -C $DIR node-v4.2.4-linux-x64/bin/node;
     else
-        #32-bit stuff goes here
-        curl -o $DIR/node.tar.gz https://nodejs.org/dist/v4.2.4/node-v4.2.4-linux-x32.tar.gz
-        echo "Unpacking node..."
-        tar --strip-components=2 -xvf $DIR/node.tar.gz -C $DIR node-v4.2.4-linux-x32/bin/node
+        #Download 32-bit node
+        curl -o $DIR/node.tar.gz https://nodejs.org/dist/v4.2.4/node-v4.2.4-linux-x32.tar.gz;
+        echo "Unpacking node...";
+        tar --strip-components=2 -xvf $DIR/node.tar.gz -C $DIR node-v4.2.4-linux-x32/bin/node;
     fi
-    runInstaller;
-    echo "Removing node tarball and extract..."
-    rm -rf $DIR/node
-    rm -f $DIR/node.tar.gz
+    runInstaller $DIR/node;
+    echo "Removing node tarball and extract...";
+    rm -rf $DIR/node;
+    rm -f $DIR/node.tar.gz;
 }
 
 if [ -z $(which node) ]; then
-    echo "Cannot find node"
-    downloadNodeAndInstall
+    echo "Cannot find node";
+    downloadNodeAndInstall;
 else
-    NODE_VERSION=$(node --version)
-    if [[ $NODE_VERSION == *"v4.2"* ]]; then
+    NODE_VERSION=$(node --version);
+    #Test if node version is v4.2.* using regex
+    if [[ $NODE_VERSION =~ ^v4\.2\..* ]]; then
```

Comment:
It's more of a stylistic choice. If we're using the double-bracket operator, it's best if we make full use of it and use of regex rather than basic wild cards. It's also easier to add checks for other node versions if you plan on supporting that in the future.